tools/ktx/command_compare.cpp

```diff
@@ -0,0 +1,2465 @@
+                    formatOptionalFileOffset(diff.fileOffsets[0], imageByteOffset, true), space,
+                    formatOptionalFileOffset(diff.fileOffsets[1], imageByteOffset, true), space, nl);
+
+                printIndent(4, "\"packed\":{}[{}", space, nl);
+                printIndent(5, "[{}{}{}],{}", space, formatPacked(texelBlockPair.first, true), space, nl);
+                printIndent(5, "[{}{}{}]{}", space, formatPacked(texelBlockPair.second, true), space, nl);
+                printIndent(4, "]");
+
+                // Only output channels if not block-compressed
+                if (!texelBlockPair.first.isBlockCompressed()) {
+                    printIndent(0, ",{}", nl);
+                    printIndent(4, "\"channels\":{}[{}", space, nl);
+                    printIndent(5, "[{}{}{}],{}", space, formatChannels(texelBlockPair.first, true), space, nl);
+                    printIndent(5, "[{}{}{}]{}", space, formatChannels(texelBlockPair.second, true), space, nl);
+                    printIndent(4, "]");
+                }
+
+                printIndent(0, "{}", nl);
+                printIndent(3, "}}");
+            }
+            if (diff.texelBlockPairList.empty()) {
+                printIndent(0, "]");
+            } else {
+                printIndent(0, "{}", nl);
+                printIndent(2, "]");
+            }
+        }
+    }
+};
+
+// -------------------------------------------------------------------------------------------------
+
+/** @page ktx_compare ktx compare
+@~English
+
+Compare two KTX2 files.
+
+@section ktx_compare_synopsis SYNOPSIS
+    ktx compare [option...] @e input-file1 @e input-file2
+
+@section ktx_compare_description DESCRIPTION
+    @b ktx @b compare compares the two KTX2 files specified as the @e input-file1 and @e input-file2
+    arguments and outputs any mismatch in texture information and/or image data.
+    The command implicitly calls @ref ktx_validate "validate" and prints any found errors
+    and warnings to stdout.
+    If any of the specified input files are invalid then comparison is done based on best effort
+    and may be incomplete.
+
+    The JSON output formats conform to the https://schema.khronos.org/ktx/compare_v0.json
+    schema even if the input file is invalid and certain information cannot be parsed or
+    displayed.
+    Additionally, for JSON outputs the KTX file identifier is printed using "\u001A" instead of
+    "\x1A" as an unescaped "\x1A" sequence inside a JSON string breaks nearly every JSON tool.
+    Note that this does not change the value of the string only its representation.
+
+    @note @b ktx @b compare prints using UTF-8 encoding. If your console is not
+    set for UTF-8 you will see incorrect characters in output of the file
+    identifier on each side of the "KTX nn".
+
+    The following options are available:
+    @snippet{doc} ktx/command.h command options_format
+    <dl>
```

Comment:
For some reason that I can't see, most of the text is hugely indented as you can see in the following. It makes reading the documentation difficult on a reasonable width terminal. Also some options have a different indent than others. I think the latter may be due to some coming from snippets.

Oh! Wait. This output is from `ktx compare --help`. I'm not sure this doxygen code is the source for this output.

```
  -f, --format text|json|mini-json
                                Specifies the report output format. Possible options are:
                                  text: Human readable text based format
                                  json: Formatted JSON
                                  mini-json: Minified JSON
                                 (default: text)
      --content raw|image|ignore
                                Controls how image content is compared. Possible values are:
                                  raw: Encoded image data is compared verbatim, as it appears in 
                                the file
                                  image: Effective image data is compared per texel block
                                  ignore: Ignore image contents
                                Note: When the mode is set to @b image, the effective image data of 
                                individual mip levels, layers, and faces is compared texel block by 
                                texel block after applying one or more of the following 
                                transformations, if needed:
                                - If the texture is supercompressed (e.g. using Zstandard or ZLIB), 
                                the images are deflated before comparison.
                                - If the texture is encoded using BasisLZ, the images are 
                                transcoded to R8G8B8A_UNORM before comparison.
                                For block compressed textures and textures encoded using UASTC the 
                                texel blocks are the individual compressed blocks, while for other 
                                formats the texel blocks are the individual pixels of the image.
                                When comparing files that use different supercompression schemes, 
                                or otherwise different encoding, additional options may need to be 
                                used to avoid unexpected differences to be reported related to the 
                                meta information of the files. For example:
                                - The supercompression scheme can be ignored with --ignore-supercomp
                                - Compressed byte length and other index section differences can be 
                                ignored with --ignore-index all or --ignore-index level
                                - DFD section differences can be ignored with --ignore-dfd all or 
                                --ignore-dfd all-except-color-space
                                - SGD section differences can be ignored with --ignore-sgd all or 
                                --ignore-sgd payload
                                 (default: raw)
      --per-pixel-output all|<number>|none
                                Controls whether per pixel / texel block difference output is 
                                generated when --content is set to image:
                                  all - Every single difference is output (may r
```

Reply:
The doxygen source and CLI help needed duplication (we still removed quite a bit of duplication in the original effort, but there's no way to merge these two).

The indentation of `--help` output is driven by cxxopts, so we do not have direct control over it.

Reply:
Is it possible to change the spacing without modifying the fmt code, e.g, a compile option, maybe in a more recent version of fmt?

I am aware of the duplication. I was not aware at first where the above text was coming from.

Reply:
This is not coming from the fmt library, but from the cxxopts library we use for CLI option handling.

I'll check if there's any possibility to configure this.

Reply:
I meant `cxxopt`. Please check. Its a more general fix than just this PR so I'm not going to hold up merging waiting for an answer.

Reply:
I've checked cxxopts. The good news is there's a single constant that affects the maximum indentation the descriptions of CLI options will have (maximum, as if all option names are shorter, then smaller indentation is used).

The bad news is that it is only modifiable by actually changing our copy of the cxxopts source code.

Do you want me to do that?

FYI, the default value is 30, meaning that the maximum indentation of descriptions is 32 (+2 because of the indentation of CLI option names themselves).